### PR TITLE
chore: replace TransactionSigned struct inits with new functions

### LINKED ISSUE #12769: Replace TransactionSigned struct inits with new functions
### Describe the feature

with https://github.com/paradigmxyz/reth/pull/12768 we have additional `new` fns to create a TransactionSigned

we need to replace all of these inits 

https://github.com/paradigmxyz/reth/blob/7d24aa40e82dab393cff77dffaf976f059153f83/crates/net/network/tests/it/big_pooled_txs_req.rs#L29-L33

with calls to the corresponding new fn

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by htiennv**
Closes #12769

### CODE REVIEW

File: crates/storage/provider/src/providers/database/provider.rs [hunk 1]
```diff
@@ -1499,11 +1496,9 @@ impl<TX: DbTx + 'static, N: NodeTypes<ChainSpec: EthereumHardforks>> Transaction
...
     fn transaction_by_hash(&self, hash: TxHash) -> ProviderResult<Option<TransactionSigned>> {
         if let Some(id) = self.transaction_id(hash)? {
-            Ok(self.transaction_by_id_no_hash(id)?.map(|tx| TransactionSigned {
-                hash: hash.into(),
-                signature: tx.signature,
-                transaction: tx.transaction,
-            }))
+            Ok(self
+                .transaction_by_id_no_hash(id)?
+                .map(|tx| TransactionSigned::new(tx.transaction, tx.signature, hash.into())))
```
Comment (mattsse) [inline crates/storage/provider/src/providers/database/provider.rs:1501]: this into is not needed I believe

File: crates/storage/provider/src/providers/database/provider.rs [hunk 2]
```diff
@@ -1517,11 +1512,8 @@ impl<TX: DbTx + 'static, N: NodeTypes<ChainSpec: EthereumHardforks>> Transaction
         let mut transaction_cursor = self.tx.cursor_read::<tables::TransactionBlocks>()?;
         if let Some(transaction_id) = self.transaction_id(tx_hash)? {
             if let Some(tx) = self.transaction_by_id_no_hash(transaction_id)? {
-                let transaction = TransactionSigned {
-                    hash: tx_hash.into(),
-                    signature: tx.signature,
-                    transaction: tx.transaction,
-                };
+                let transaction =
+                    TransactionSigned::new(tx.transaction, tx.signature, tx_hash.into());
```
Comment (mattsse) [inline crates/storage/provider/src/providers/database/provider.rs:1516]: same

**Review by mattsse — APPROVED** [review]
lgtm,pending lint

**Comment by htiennv** [comment]
Hi @mattsse , i'm just force pushed for resolve conflicts. Please check again, thanks!

**Review by mattsse — APPROVED** [review]
nice,ty

**Pull request merged**